tools/scripts/list_stl_dependencies.sh

```diff
@@ -22,30 +22,74 @@
 #   ./list_stl_dependencies.sh
 #=====================================================
 
+SCOPE=${1:-list}
 COMPONENTS=(iceoryx_hoofs iceoryx_posh)
 SOURCE_DIR=(source include)
 WORKSPACE=$(git rev-parse --show-toplevel)
+QNX_PLATFORM_DIR=$WORKSPACE/iceoryx_hoofs/platform/qnx/
+USELIST=$WORKSPACE/tools/scripts/used-headers.txt
+TEMP_FILE=$(mktemp)
+GET_HEADER_NAME="\<\K[^<>]+(?=>)" # Matches the content between angle brackets
 
 for COMPONENT in ${COMPONENTS[@]}; do
     for DIR in ${SOURCE_DIR[@]}; do
         GREP_PATH="${GREP_PATH} ${WORKSPACE}/${COMPONENT}/$DIR"
     done
 done
 
+echo "# QNX platform/ libc headers" >> $TEMP_FILE
+
+echo "# usage of headers for QNX"
+gcc -w -fpreprocessed -dD -E $(find $QNX_PLATFORM_DIR -type f -iname *.cpp -o -iname *.hpp) \
+ | grep -e "#include <" \
+ | grep -oP $GET_HEADER_NAME \
+ | sort \
+ | uniq \
+ | tee -a $TEMP_FILE \
+ | cat
+
+
+echo "# iceoryx_posh / iceoryx_hoofs headers" >> $TEMP_FILE
+# GCC can't preprocess .inl so we grep them in plain text
 echo
-echo usage of std components by file
-grep -RIne "std::" $GREP_PATH | sed -n  "s/\([^:]*\:[0-9]*\)\:.*\(std::[a-zA-Z_]*\).*/\ \ \1  \2/p" | sort | uniq
+echo "# usage of <..> header includes"
+{ gcc -w -fpreprocessed -dD -E $(find $GREP_PATH -type f -iname *.cpp -o -iname *.hpp); cat $(find $GREP_PATH -type f -iname *.inl); } \
+ | grep -e "#include <" \
+ | grep -oP $GET_HEADER_NAME \
+ | sort \
+ | uniq \
+ | tee -a $TEMP_FILE \
+ | cat
+
+if [[ "$SCOPE" == "check" ]]; then
+    echo
+    echo "Comparing the used system headers against the list.."
+    diff $TEMP_FILE $USELIST
+    if [ $? -eq 1 ]; then
+        echo "One or more header is not on the list, please remove the header usage or extend the list!"
+        exit 1
+    fi
+    echo "No header divergence found!"
+fi
 
 echo
-echo files with stl dependency
+echo "# usage of std components"
+{ gcc -w -fpreprocessed -dD -E $(find $GREP_PATH -type f -iname *.cpp -o -iname *.hpp); cat $(find $GREP_PATH -type f -iname *.inl); } \
+ | grep -HIne "std::" \
+ | sed -n  "s/\([^:]*\:[0-9]*\)\:.*\(std::[a-zA-Z_]*\).*/\ \  \2/p" \
+ | sort \
+ | uniq
+
+echo
+echo "# files with stl dependency"
 grep -RIne "std::" $GREP_PATH | sed -n  "s/\([^:]*\)\:[0-9]*\:.*std::[a-zA-Z_]*.*/\1/p" | xargs -I{} basename {} | sort | uniq
 
 echo
-echo using namespace with std component
+echo "# using namespace with std component"
 grep -RIne ".*using[ ]*namespace[ ]*std" $GREP_PATH | sed -n "s/\(.*\)/\ \ \1/p"
```

Comment:
Please use `\s*` instead of `[ ]*`. Clang-format should replace all tabs with space and the compiler should issue a warning when something like no-break space is used but `\s` reduces the risk.

Additionally, the regex does not find `using namespace ::std;`.

Reply:
True indeed, replaced. As this is just for informational purposes currently I think we can live without `using namespace ::std;`